### ISSUE TITLE
KeyError on Hardhat directory 

### ISSUE THREAD

**Issue by Evert0x** [crytic/slither#860]
I'm trying to run Slither on this repo [`sherlock-v1-core`](https://github.com/sherlock-protocol/sherlock-v1-core). 

I'm doing `slither .` in the root of the hardhat directory, using slither 0.8.0 (`4b5583944`). 

Am I doing something wrong? This is the exception that follows.

```
(venv) evert@EvertX1:~/sherlock/sherlock-v1-core$ slither .
'npx hardhat compile --force' running
Compiling 49 files with 0.7.6
Compilation finished successfully

~ compile warnings removed ~

Traceback (most recent call last):
  File "/home/evert/slither/slither/__main__.py", line 723, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/evert/slither/slither/__main__.py", line 82, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/home/evert/slither/slither/__main__.py", line 65, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/home/evert/slither/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/home/evert/slither/slither/solc_parsing/slither_compilation_unit_solc.py", line 171, in parse_top_level_from_loaded_json
    contract.set_offset(top_level_data["src"], self._compilation_unit)
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 102, in _convert_source_mapping
    compilation_unit, filename, s, l
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 44, in _compute_line
    filename, start + length
  File "/home/evert/slither/venv/lib/python3.6/site-packages/crytic_compile/crytic_compile.py", line 233, in get_line_from_offset
    return lines_delimiters[offset]
KeyError: 7421
ERROR:root:None
ERROR:root:Error in .
ERROR:root:Traceback (most recent call last):
  File "/home/evert/slither/slither/__main__.py", line 723, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/evert/slither/slither/__main__.py", line 82, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/home/evert/slither/slither/__main__.py", line 65, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/home/evert/slither/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/home/evert/slither/slither/solc_parsing/slither_compilation_unit_solc.py", line 171, in parse_top_level_from_loaded_json
    contract.set_offset(top_level_data["src"], self._compilation_unit)
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 102, in _convert_source_mapping
    compilation_unit, filename, s, l
  File "/home/evert/slither/slither/core/source_mapping/source_mapping.py", line 44, in _compute_line
    filename, start + length
  File "/home/evert/slither/venv/lib/python3.6/site-packages/crytic_compile/crytic_compile.py", line 233, in get_line_from_offset
    return lines_delimiters[offset]
KeyError: 7421
```

**Comment by arbazkiraak** [comment]
Facing a similar problem with the latest solidity version.

**Comment by suranap** [comment]
I had this same problem but fixed it just now. Delete artifacts/, typechain/, and cache/. Ran `slither .` again and it seems to work.

**Comment by 0xPoseidon** [comment]
> I had this same problem but fixed it just now. Delete artifacts/, typechain/, and cache/. Ran `slither .` again and it seems to work.

This workaround does not reliably work on Windows, Hardhat 2.5.x and solc 0.8.6

**Comment by montyly** [comment]
Hi, @Evert0x, thanks for reporting this.

I believe we fixed it-  it works with `master`, and the fix will be available in the upcoming release https://github.com/crytic/slither/issues/905

@arbazkiraak, @0xPoseidon : can you point me to the code that is failing? I will try slither on it before the next release

**Comment by montyly** [comment]
We released Slither 0.8.1 yesterday and, unless I am mistaken, this is now fixed

**Comment by meetmangukiya** [comment]
Still seems to be happening on 0.8.1. Similar error:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 741, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 83, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 66, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 171, in parse_top_level_from_loaded_json
    contract.set_offset(top_level_data["src"], self._compilation_unit)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 101, in _convert_source_mapping
    (lines, starting_column, ending_column) = self._compute_line(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 43, in _compute_line
    end_line, ending_column = compilation_unit.core.crytic_compile.get_line_from_offset(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/crytic_compile/crytic_compile.py", line 284, in get_line_from_offset
    return lines_delimiters[offset]
KeyError: 11747
None
Error in .
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 741, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 83, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/__main__.py", line 66, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 171, in parse_top_level_from_loaded_json
    contract.set_offset(top_level_data["src"], self._compilation_unit)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 101, in _convert_source_mapping
    (lines, starting_column, ending_column) = self._compute_line(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/slither/core/source_mapping/source_mapping.py", line 43, in _compute_line
    end_line, ending_column = compilation_unit.core.crytic_compile.get_line_from_offset(
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/crytic_compile/crytic_compile.py", line 284, in get_line_from_offset
    return lines_delimiters[offset]
KeyError: 11747
```

Complete logs: https://github.com/ethereum-optimism/optimism/runs/4066860066 check the `Run analysis` step.
